servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/NettyChannelPublisher.java

```diff
@@ -318,14 +333,44 @@ private void assertInEventloop() {
 
     private final class SubscriptionImpl implements Subscription {
 
+        @Nullable
+        private final ReadObserver observer;
         final Subscriber<? super T> associatedSub;
 
-        private SubscriptionImpl(Subscriber<? super T> associatedSub) {
-            this.associatedSub = associatedSub;
+        private SubscriptionImpl(final Subscriber<? super T> subscriber,
+                                 @Nullable final NonMultiplexedObserver readWriteObserver) {
+            observer = readWriteObserver == null ? null : requireNonNull(readWriteObserver.onNewRead());
+            this.associatedSub = observer == null ? subscriber : new Subscriber<T>() {
```

Comment:
this seems backward to wrap the `Subscriber` inside a `Subscription`. Also, we are not handling errors thrown from the observer.
Can we start with using `when*` methods on the `NettyChannelPublisher` for this instrumentation instead of baking it in here?